main.cpp

```diff
@@ -0,0 +1,90 @@
+#include <iostream>
+using std::cin;
+using std::cout;
+using std::endl;
+
+#include<cmath>
+using std::abs;
+
+int gcd_iterative(int a, int b){
+	a = abs(a);
+	b = abs(b);
+
+	while(a != 0 || b != 0){
+		if(a == b){
+			return a;
+		}
+		if(a > b){
+			a -= b;
+		}else{
+			b -= a;
+		}
+	}
+
+	return (a == 0)? a : b;
+}
+
+//recursive version
+int gcd(int a, int b){
+	a = abs(a);
+	b = abs(b);
+	if(a == 0 || b == 0){
+		return a + b;
+	}
+
+	if(a > b){
+		return gcd(a-b,b);
+	}
+	
+	if(a < b){
+		return gcd(a, b-a);
+	}
+
+	if(a == b){
+		return a;
+	}
+
+}
+
+int fibonacci_iterative(int n, int f1 = 1, int f2 = 1){
+	if(n <= 2) {
+		return 1;
+	}
```

Comment:
this assumes that `f1` and `f2` will always both be 1 (which is valid, if you're not trying to generalize the sequence) -- in which case, they should probably be local variables rather than function parameters with default arguments.